code/Controllers/CMSMain.php

```diff
@@ -462,8 +465,9 @@ public function getSiteTreeFor(
         $childrenMethod = null,
         $numChildrenMethod = null,
         $filterFunction = null,
-        $nodeCountThreshold = 30
+        $nodeCountThreshold = null
     ) {
+        $nodeCountThreshold = is_null($nodeCountThreshold) ? Config::inst()->get(get_class(), 'default_node_count_threshold') : $nodeCountThreshold;
```

Comment:
Instead of adding a new config to CMSMain (new API) can we use the SiteTree config? This is pre-existing and means fewer changes CMS developers need to make to update a hierarchy's behaviour.

```
if (!isset($nodeCountThreshold)) {
	$nodeCountThreshold = Config::inst()->get($className, 'node_threshold_total');
}
```